examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -532,6 +532,9 @@ static void wfx_rsi_do_join(void)
             break;
         case WFX_SEC_WPA3:
             connect_security_mode = RSI_WPA3;
+#if WIFI_ENABLE_SECURITY_MODE_WPA3_TRANSITION
```

Comment:
Still believe this is not required. Better to have only 2 
1. WPA3 transition
2. WPA/WPA2 mixed mode

Since WPA3 only can be handled from the WPA3 transition mode